src/features/locators/utils/locatorValidation.ts

```diff
@@ -32,15 +34,14 @@ export const validateXpath = (
       const err = `${length} ${LocatorValidationErrors.MultipleElements}` as LocatorValidationErrorType; //validationStatus: ERROR;
       throw new Error(err);
     } else if (length === 1) {
-      if (foundHash !== _jdnHash) {
-        if (equalHashes(foundHash, locators).length) {
-          throw new Error(LocatorValidationErrors.DuplicatedLocator); //validationStatus: ERROR
-        } else {
-          const msg = isCreatingForm
-            ? "" //validationStatus: SUCCESS
-            : LocatorValidationWarnings.NewElement; //validationStatus: WARNING
-          return msg;
-        }
+      const duplicates = equalHashes(foundHash, locators).filter((locator) => locator.element_id !== _element_id);
```

Comment:
Can you pls explain why we filter locators twice? by hash at first time, and by element_id at second.
Can't we filter one time by element_id and validation message?